Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialTypeSourceData.cpp

```diff
@@ -112,89 +127,351 @@ namespace AZ
+
+                for (auto& property : propertyGroup->m_properties)
                 {
-                    if (property.m_name == propertyName)
+                    if (!callback(propertyNameContext2, property.get()))
                     {
-                        return &property;
+                        return false; // Stop processing
                     }
                 }
+
+                if (!EnumerateProperties(callback, propertyNameContext2, propertyGroup->m_propertyGroups))
+                {
+                    return false; // Stop processing
+                }
             }
 
-            // Property has not been found, try looking for renames in the version history
+            return true;
+        }
 
-            MaterialPropertyId propertyId = MaterialPropertyId{groupName, propertyName};
-            ApplyPropertyRenames(propertyId);
+        bool MaterialTypeSourceData::EnumerateProperties(const EnumeratePropertiesCallback& callback) const
+        {
+            if (!callback)
+            {
+                return false;
+            }
 
-            // Do the search again with the new names
+            return EnumerateProperties(callback, {}, m_propertyLayout.m_propertyGroups);
+        }
 
-            AZStd::vector<AZStd::string> tokens;
-            AZ::StringFunc::Tokenize(propertyId.GetStringView(), tokens, ".", true, true);
-            if (tokens.size() == 2)
+        bool MaterialTypeSourceData::ConvertToNewDataFormat()
+        {            
+            for (const auto& group : GetOldFormatGroupDefinitionsInDisplayOrder())
             {
-                groupIter = m_propertyLayout.m_properties.find(tokens[0]);
-                if (groupIter != m_propertyLayout.m_properties.end())
+                auto propertyListItr = m_propertyLayout.m_propertiesOld.find(group.m_name);
+                if (propertyListItr != m_propertyLayout.m_propertiesOld.end())
                 {
-                    for (const PropertyDefinition& property : groupIter->second)
+                    const auto& propertyList = propertyListItr->second;
+                    for (auto& propertyDefinition : propertyList)
                     {
-                        if (property.m_name == tokens[1])
+                        PropertyGroup* propertyGroup = FindPropertyGroup(group.m_name);
+
+                        if (!propertyGroup)
                         {
-                            return &property;
+                            m_propertyLayout.m_propertyGroups.emplace_back(AZStd::make_unique<PropertyGroup>());
+                            m_propertyLayout.m_propertyGroups.back()->m_name = group.m_name;
+                            m_propertyLayout.m_propertyGroups.back()->m_displayName = group.m_displayName;
+                            m_propertyLayout.m_propertyGroups.back()->m_description = group.m_description;
+                            propertyGroup = m_propertyLayout.m_propertyGroups.back().get();
```

Comment:
[suggestion] create and setup the unique pointer first, then add it to the vector. This avoids all the accessed to members and backs.

````
auto newProperyGroup = AZStd::make_unique<PropertyGroup>();
newProperyGroup->m_name = group.m_name;
newProperyGroup->m_displayName = group.m_displayName;
newProperyGroup->m_description = group.m_description;

m_propertyLayout.m_propertyGroups.push_back(AZStd::move(newProperyGroup ));
propertyGroup = m_propertyLayout.m_propertyGroups.back().get();
````